### PR TITLE
STORM-1292: port backtype.storm.messaging-test to java View changes

### PR DESCRIPTION
**Pull request by kamleshbhatt**
This PR is in continuation of https://github.com/apache/storm/pull/1839.

I have applied all review comments in this PR.

### CODE REVIEW

**Comment by harshach** [comment]
@kamleshbhatt this looks good to me. Can you please squash the commits into single one.. After that I am +1.

**Comment by kamleshbhatt** [comment]
@harshach thanks for reviewing the PR. I tried to squash the commits but not sure why it is throwing error.

**Comment by harshach** [comment]
@kamleshbhatt  you can do the following in your branch.
git rebase -i HEAD~9
and pick one commit as your main commit and add "s" against all the other commits.
and change the commit title to "STORM-1292: port backtype.storm.messaging-test to java View changes"
git push -f origin master to push the changes into your repo which will be reflected in the PR automatically.

I recommend you to branch your repo for any contributions . Usually I keep the master on par with apache storm master and don't make any changes in the maser.
if you are working on a JIRA . Do git checkout -b STORM-1292 and make changes within the branch and push to your fork. This way you can keep the master clean and sync up with apache storm master.

File: storm-core/test/jvm/org/apache/storm/MessagingTest.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        Config stormConf = new Config();
+        stormConf.put(Config.TOPOLOGY_WORKERS, 2);
+        stormConf.put(Config.STORM_MESSAGING_TRANSPORT , "org.apache.storm.messaging.netty.Context");
+
+        try (ILocalCluster cluster = new LocalCluster.Builder().withSimulatedTime()
+                    .withSupervisors(1).withPortsPerSupervisor(2)
+                    .withDaemonConf(stormConf).build()) {
+
+            TopologyBuilder builder = new TopologyBuilder();
+            builder.setSpout("1", new TestWordSpout(false), 2);
```
Comment (HeartSaVioR) [inline storm-core/test/jvm/org/apache/storm/MessagingTest.java]: false -> true

**Comment by HeartSaVioR** [comment]
@kamleshbhatt Thanks for the work. Left a minor comment. If you have hard time to rebase please address my comment and leave a comment to help. I'll handle it.

**Comment by kamleshbhatt** [comment]
@harshach I tried your suggestion "git rebase -i HEAD~9" and kept just one comment and put s in front of all other comments. It gives error
Cannot 'squash' without a previous commit

I liked your idea to work on branch rather than master. I'll follow this from next task. Thanks for the help.

**Comment by kamleshbhatt** [comment]
@HeartSaVioR Applied your review comment. However not able to squash the commits. Please help on this.
Once again thanks a lot for your help, always.

**Comment by HeartSaVioR** [comment]
@kamleshbhatt 
I can't rebase either so cherry-picked some valid commits and rebased myself. Thanks for the work.

**Pull request closed without merging**